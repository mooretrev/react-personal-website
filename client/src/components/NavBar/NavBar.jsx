import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import CssBaseline from '@material-ui/core/CssBaseline';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import useTheme from '@material-ui/core/styles/useTheme';
import HomeIcon from '@material-ui/icons/Home';
import FastfoodIcon from '@material-ui/icons/Fastfood';
import FitnessCenterIcon from '@material-ui/icons/FitnessCenter';
import FavoriteIcon from '@material-ui/icons/Favorite';
import { Link } from 'react-router-dom';
import { useAuth0 } from '@auth0/auth0-react';
import useStyles from '../../styles/Index.jsx';
import LoginButton from '../LoginButton.jsx';
import LogoutButton from '../LogoutButton.jsx';

function NavBar(props) {
  const classes = useStyles();
  const theme = useTheme();
  const [mobileOpen, setMobileOpen] = React.useState(false);
  const [title, setTitle] = useState('Home');

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  const renderSwitch = (index) => {
    switch (index) {
      case 0:
        return <HomeIcon />;
      case 1:
        return <FastfoodIcon />;
      case 2:
        return <FitnessCenterIcon />;
      default:
    }
    return <div />;
  };

  const createLocation = (text) => {
    const location = text.toLowerCase();
    if (location === 'home') {
      return '/';
    }
    return (`/${text.toLowerCase()}`);
  };

  const handleNavClick = (text) => {
    setTitle(text);
  };

  const authenticationButton = () => {
    const { isAuthenticated } = useAuth0();

    return isAuthenticated ? <LogoutButton /> : <LoginButton />;
  };

  const drawer = (
    <div>
      <div className={classes.drawerTitle}>
        <Typography variant="h5">Trevor Moore</Typography>
      </div>
      <div>
        <Divider />
        <List>
          {['Home', 'Recipes', 'Powerlifting'].map((text, index) => (
            <Link className={classes.link} to={createLocation(text)} key={text}>
              <ListItem button onClick={() => handleNavClick(text)}>
                <ListItemIcon>{renderSwitch(index)}</ListItemIcon>
                <ListItemText primary={text} />
              </ListItem>
            </Link>
          ))}
        </List>
        <Divider />
        <List>
          <Link className={classes.link} to="/risheil" key="rishiel">
            <ListItem button>
              <ListItemIcon><FavoriteIcon /></ListItemIcon>
              <ListItemText primary="Risheil" />
            </ListItem>
          </Link>
        </List>

      </div>
    </div>
  );

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={handleDrawerToggle}
            className={classes.menuButton}
          >
            <MenuIcon />
          </IconButton>
          <Grid container>
            <Grid item xs={8}>
              <Typography variant="h6" noWrap>
                {title}
              </Typography>
            </Grid>
            <Grid item container xs={4} justify="flex-end">
              {authenticationButton()}
            </Grid>
          </Grid>

        </Toolbar>
      </AppBar>
      <nav className={classes.drawer}>
        <Hidden smUp implementation="css">
          <Drawer
            variant="temporary"
            anchor={theme.direction === 'rtl' ? 'right' : 'left'}
            open={mobileOpen}
            onClose={handleDrawerToggle}
            classes={{
              paper: classes.drawerPaper,
            }}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
          >
            {drawer}
          </Drawer>
        </Hidden>
        <Hidden xsDown implementation="css">
          <Drawer
            classes={{
              paper: classes.drawerPaper,
            }}
            variant="permanent"
            open
          >
            {drawer}
          </Drawer>
        </Hidden>
      </nav>
    </div>
  );
}

export default NavBar;
