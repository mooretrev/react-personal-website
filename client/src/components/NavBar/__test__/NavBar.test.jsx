import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../NavBar.jsx';

describe('<NavBar />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavBar window={0} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
