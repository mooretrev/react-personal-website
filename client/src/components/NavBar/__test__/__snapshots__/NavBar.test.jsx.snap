// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-3"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={8}
        >
          <WithStyles(ForwardRef(Typography))
            noWrap={true}
            variant="h6"
          >
            Home
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          item={true}
          justify="flex-end"
          xs={4}
        >
          <LoginButton />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <nav
    className="makeStyles-drawer-2"
  >
    <Hidden
      implementation="css"
      smUp={true}
    >
      <WithStyles(ForwardRef(Drawer))
        ModalProps={
          Object {
            "keepMounted": true,
          }
        }
        anchor="left"
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-7",
          }
        }
        onClose={[Function]}
        open={false}
        variant="temporary"
      >
        <div>
          <div
            className="makeStyles-drawerTitle-6"
          >
            <WithStyles(ForwardRef(Typography))
              variant="h5"
            >
              Trevor Moore
            </WithStyles(ForwardRef(Typography))>
          </div>
          <div>
            <WithStyles(ForwardRef(Divider)) />
            <WithStyles(ForwardRef(List))>
              <Link
                className="makeStyles-link-12"
                key="Home"
                to="/"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <HomeIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Home"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Recipes"
                to="/recipes"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FastfoodIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Recipes"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Powerlifting"
                to="/powerlifting"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FitnessCenterIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Powerlifting"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Stock"
                to="/stock"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <ShowChartIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Stock"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
            </WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(Divider)) />
            <WithStyles(ForwardRef(List))>
              <Link
                className="makeStyles-link-12"
                key="rishiel"
                to="/risheil"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FavoriteIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Risheil"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
            </WithStyles(ForwardRef(List))>
          </div>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
    <Hidden
      implementation="css"
      xsDown={true}
    >
      <WithStyles(ForwardRef(Drawer))
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-7",
          }
        }
        open={true}
        variant="permanent"
      >
        <div>
          <div
            className="makeStyles-drawerTitle-6"
          >
            <WithStyles(ForwardRef(Typography))
              variant="h5"
            >
              Trevor Moore
            </WithStyles(ForwardRef(Typography))>
          </div>
          <div>
            <WithStyles(ForwardRef(Divider)) />
            <WithStyles(ForwardRef(List))>
              <Link
                className="makeStyles-link-12"
                key="Home"
                to="/"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <HomeIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Home"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Recipes"
                to="/recipes"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FastfoodIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Recipes"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Powerlifting"
                to="/powerlifting"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FitnessCenterIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Powerlifting"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
              <Link
                className="makeStyles-link-12"
                key="Stock"
                to="/stock"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <ShowChartIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Stock"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
            </WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(Divider)) />
            <WithStyles(ForwardRef(List))>
              <Link
                className="makeStyles-link-12"
                key="rishiel"
                to="/risheil"
              >
                <WithStyles(ForwardRef(ListItem))
                  button={true}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <FavoriteIcon />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    primary="Risheil"
                  />
                </WithStyles(ForwardRef(ListItem))>
              </Link>
            </WithStyles(ForwardRef(List))>
          </div>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
  </nav>
</div>
`;
